### ISSUE TITLE
`bincode` breaks `Enums`

### ISSUE THREAD

**Issue by KuSpa** [CabbitStudios/bevy_spicy_networking#12]
The change from `JSON` to `bincode` introduced [here](https://github.com/CabbitStudios/bevy_spicy_networking/commit/59779cfdb025fb6d72e0b8539674f7d72f9778e6) breaks Enum transmissions. Upon receiving one gets the error 
```
ERROR bevy_spicy_networking::client: Failed to decode network packet from [127.0.0.1:9999]: Bincode does not support Deserializer::deserialize_identifier
```

I set up a small example for verification [here](https://github.com/KuSpa/not_so_spicy_enums).
## Reproduce
run
```
cargo run --package server
```
run
```
cargo run --package client
```
click on the grey `server` bevy app. Every click broadcasts an `MyEnum::A` wich cannot be parsed from the client.

**Comment by TheNeikos** [comment]
You are correct, bincode has some weird restrictions as it is a self-describing format and thus can't work with things like that.

I do consider the use of bincode in that context sub-par, but couldn't find a suitable one. I am open for suggestions!

**Comment by KuSpa** [comment]
I don't know whether this is a limitation of `bincode`, as they support `Enums`, which I verified [here](https://gist.github.com/KuSpa/3492ef1f8faaf4b2576fa3cc34a5f333).

**Comment by TheNeikos** [comment]
Yes, as it is able to differentiate the two. As you can see in this issue, this problem is known and probably will not be fixed: https://github.com/bincode-org/bincode/issues/272#issuecomment-562872027

I am open for alternate defaults. I plan on adding compile-time picking of encoding scheme, but did not get around to do that yet.

**Comment by benfalk** [comment]
I have a rough compile-time encoding scheme plan in my fork, I need to get it into a better place though before I do and then I'll submit a PR for it, been kicking the tires on it with this server mmo project and I've made way to many other changes that I should roll into a separate performance PR
